src/compiler/crystal/semantic/restrictions.cr

```diff
@@ -1339,68 +1339,72 @@ module Crystal
     end
   end
 
-  class NumberAutocastType
+  class AutocastType
```

Comment:
It seems like a good case for an `abstract` class.

Reply:
That's already the case: it's defined in `types.cr`